sktime/transformations/series/date.py

```diff
@@ -255,11 +257,13 @@ def _calendar_dummies(x, funcs):
         values = (
             (x["date_sequence"] - quarter_start) / pd.to_timedelta("1D") + 1
         ).astype(int)
-        x[funcs] = values
-        return x[funcs]
+        cd = values
     else:
-        x[funcs] = getattr(date_sequence, funcs)
-        return x[funcs]
+        cd = getattr(date_sequence, funcs)
+    cd = pd.DataFrame(cd)
+    cd.rename(columns={cd.columns[0]: funcs}, inplace=True)
```

Comment:
let's avoid `inplace=True`, that may change in futuer `pandas` versions

Reply:
done